log/brimapi/logger.go

```diff
@@ -60,3 +73,22 @@ func (lh *LogHook) Fire(entry *logrus.Entry) error {
 	}
 	return discardBody(resp)
 }
+
+// Discovery service URI for service:// protocol
+func (lh *LogHook) discoveryServiceURI(URI string) (*url.URL, error) {
+	parsedURI, err := url.Parse(URI)
+	if err != nil {
+		return nil, err
+	}
+
+	if !strings.HasPrefix(parsedURI.Scheme, serviceDiscoverySchemeName) {
+		return parsedURI, nil
+	}
+
+	consulClient, err := api.NewClient(api.DefaultConfig())
+	if err != nil {
+		return nil, fmt.Errorf("unable to create Consul client: %s", err)
+	}
+	resolver := service.New(consulClient)
+	return resolver.GetRandomNode(parsedURI.Host)
```

Comment:
This method actually doest a lot of stuff that I didn't expect after just seeing the signature. There is an implicit setup of consul client, it should be more verbose